src/2dplane/GridStateSpace.cpp

```diff
@@ -10,12 +9,39 @@ using namespace std;
 GridStateSpace::GridStateSpace(float width, float height, int discretizedWidth, int discretizedHeight):
     PlaneStateSpace(width, height),
     _obstacleGrid(width, height, discretizedWidth, discretizedHeight) {
+    setMinStepSize(.1);
+    setMaxStepSize(5);
+    setDistScale(1);
 }
 
 bool GridStateSpace::stateValid(const Vector2f &pt) const {
     return PlaneStateSpace::stateValid(pt) && !_obstacleGrid.obstacleAt(_obstacleGrid.gridSquareForLocation(pt));
 }
 
+Vector2f GridStateSpace::intermediateState(const Vector2f &source, const Vector2f &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const {
+    float stepSize = prevStepSize;
+    Vector2f delta = target - source;
+    delta = delta / delta.norm();   //  unit vector
+    if (prevStepSize == 0) {
+        stepSize = defaultStepSize;
+    } else {
+        float dist = _obstacleGrid.nearestObstacleDist(source, maxStepSize());
+        if (dist > prevStepSize * distScale()) { //grows if dist > max tolerable dist
```

Comment:
The fact that `distScale` is being multiplied by `prevStepSize` here means that the threshold for growing or shrinking is constantly changing.  Is this what we want?